portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +34,59 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+//   private ArrayList<String> comments = new ArrayList<String>();
+  
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {    
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
```

Comment:
No need to act on this, just wanted to show you a cool Java feature. I learned this at work and my mind was blown.

Java 8 introduced this very cool functional programming feature called lambdas, which enabled a second very cool programming paradigm called Streams. Using a stream, this expression could be rewritten as:

ArrayList<Comment> comments = results.asIterable().stream().map(Comment::fromEntity).collect(Collectors.toCollection(ArrayList::new));

More about streams:
https://www.tutorialspoint.com/java8/java8_streams.htm

More about the colons (method reference):
https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html

More about lambdas:
https://www.geeksforgeeks.org/lambda-expressions-java-8/

Reply:
Whoa, that's some really intense syntax. I'll definitely read more into it!